google/cloud/aiplatform/pipeline_service/pipeline_based_service.py

```diff
@@ -0,0 +1,202 @@
+    _getter_method = "get_pipeline_job"
+    _list_method = "list_pipeline_jobs"
+    _parse_resource_name_method = "parse_pipeline_job_path"
+    _format_resource_name_method = "pipeline_job_path"
+
+    _valid_done_states = _PIPELINE_COMPLETE_STATES
+
+    @property
+    @abc.abstractmethod
+    def backing_pipeline_job(self) -> pipeline_jobs.PipelineJob:
+        """The PipelineJob associated with the resource."""
+        pass
+
+    @property
+    @abc.abstractmethod
+    def pipeline_console_uri(self) -> str:
+        """The console URI of the PipelineJob created by the service."""
+        pass
+
+    @property
+    @abc.abstractmethod
+    def metadata_output_artifact(self) -> Optional[str]:
+        """The ML Metadata output artifact resource URI from the completed pipeline run."""
+        pass
+
+    @property
+    def state(self) -> Optional[str]:
+        """The state of the Pipeline run associated with the service."""
+        if self.backing_pipeline_job:
+            return self.backing_pipeline_job.state
+        return None
+
+    @classmethod
+    @abc.abstractmethod
+    def _template_ref(self) -> str:
+        """The pipeline template URL for this service."""
+        pass
+
+    def __init__(
+        self,
+        pipeline_job_id: str,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Retrieves an existing Pipeline Based Service given the ID of the pipeline execution.
+
+        Example Usage:
+
+            pipeline_service = aiplatform.VertexAiPipelineBasedService(
+                pipeline_job_id = "projects/123/locations/us-central1/pipelinesJobs/456"
+            )
+
+            pipeline_service = aiplatform.VertexAiPipelinebasedService(
+                pipeline_job_id = "456"
+            )
+
+        Args:
+            pipeline_job_id(str):
+                Required. A fully-qualified pipeline job run ID.
+                Example: "projects/123/locations/us-central1/pipelineJobs/456" or
+                "456" when project and location are initialized or passed.
+            project (str):
+                Optional. Project to retrieve pipeline job from. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional. Location to retrieve pipeline job from. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to retrieve this pipeline job. Overrides
+                credentials set in aiplatform.init.
+        """
+
+        super().__init__(
+            project=project,
+            location=location,
+            credentials=credentials,
+            resource_name=pipeline_job_id,
+        )
+        self._gca_resource = self._get_gca_resource(resource_name=pipeline_job_id)
```

Comment:
Note that this should have a check to see if the pipeline template used in the PipelineJob is consistent with the subclass `_template_ref` and raise if not.